build-tools/packages/build-cli/src/library/changesets.ts

```diff
@@ -3,86 +3,262 @@
+}
+
 /**
  * A ChangesetEntry is an object containing flattened content and file metadata from a changesets file. Changeset files
  * themselves include a mapping of package to version bump type. This object includes the change type and a single
  * package, effectively flattening the changesets.
  */
-export type ChangesetEntry = Omit<Changeset, "metadata" | "changeTypes"> & {
+export type ChangesetEntry = Omit<Changeset, "metadata" | "mainPackage" | "changeTypes"> & {
+	/**
+	 * The name of the package this ChangesetEntry applies to.
+	 */
 	pkg: string;
+
+	/**
+	 * This will be true if the package in this ChangesetEntry is the main package for the changeset.
+	 */
+	isMainPackage: boolean;
+
+	/**
+	 * The type of release this changeset represents.
+	 */
 	changeType: VersionBumpType;
+
+	/**
+	 * The original full changeset that was tthe source for this ChangesetEntry.
+	 */
+	fullChangeset: Readonly<Changeset>;
 };
 
-function compareChangesets<T extends Pick<Changeset, "added">>(a: T, b: T): number {
-	if (a.added === undefined || b.added === undefined) {
+/**
+ * Compares a changeset by the highlight property of additional metadata if present, then by commit date.
+ */
+function compareChangesets<T extends Pick<Changeset, "commit" | "additionalMetadata">>(
+	a: T,
+	b: T,
+): number {
+	// Sort highlighted items to the top;
+	if (a.additionalMetadata?.highlight === true && b.additionalMetadata?.highlight !== true) {
+		return -1;
+	}
+	if (a.additionalMetadata?.highlight !== true && b.additionalMetadata?.highlight === true) {
+		return 1;
+	}
+
+	// Finally sort by date
+	if (a.commit?.date === undefined || b.commit?.date === undefined) {
 		return 0;
 	}
-	return compareAsc(a.added, b.added);
+	return compareAsc(a.commit?.date, b.commit?.date);
 }
 
 /**
- * Loads changeset files into a list of individual changes for a package. Changeset files themselves include a mapping
- * of package to version bump type. This function returns an array with an entry per-package-changeset, effectively
- * flattening the changesets.
+ * Loads changeset files into an array of {@link Changeset}s.
  *
  * @param dir - The directory containing changesets.
  * @param log - An optional logger.
- * @returns An array containing the flattened changesets.
+ * @returns An array containing the changesets.
  */
 export async function loadChangesets(dir: string, log?: Logger): Promise<Changeset[]> {
 	const repo = new Repository({ baseDir: dir });
 	const changesetFiles = await globby(["*.md", "!README.md"], { cwd: dir, absolute: true });
 	const changesets: Changeset[] = [];
 
 	for (const file of changesetFiles) {
+		log?.verbose(`Loading changeset: ${file}`);
+
 		// Get the date the changeset file was added to git.
 		// eslint-disable-next-line no-await-in-loop
-		const results = await repo.gitClient.log({ file, maxCount: 1, strictDate: true });
+		const results = await repo.gitClient.log({ file, strictDate: true });
+		const rawCommit = results.all?.at(-1);
```

Comment:
Just for my understanding, this is looking at the last entry returned by `git log`, so the earliest commit in which the changeset file appears, right?

Reply:
Yeah - clarified in a comment too.

Reply:
This was actually a pre-existing bug - the ordering has never been correct because the commits (and thus the dates) were wrong.